tests/integration_test/action_service_test.py

```diff
@@ -5472,6 +5472,791 @@ def _run_action(*args, **kwargs):
+                {"event": "bot", "timestamp": 1594907117.04548, "text": "can not = can't",
+                 "data": {"elements": None, "quick_replies": None, "buttons": None, "attachment": None,
+                          "image": None, "custom": None}, "metadata": {}}],
+            "latest_input_channel": "rest",
+            "active_loop": {},
+            "latest_action": {},
+        },
+        "domain": {
+            "config": {},
+            "session_config": {},
+            "intents": [],
+            "entities": [],
+            "slots": {"bot": "5f51zd0a56b698ca10d35d2e"},
+            "responses": {},
+            "actions": [],
+            "forms": {},
+            "e2e_actions": []
+        },
+        "version": "version"
+    }
+    with mock.patch.dict(Utility.environment, {'web_search_url': {"url": search_engine_url}}):
+        response = client.post("/webhook", json=request_object)
+        response_json = response.json()
+        assert response.status_code == 200
+        assert response_json == {'events': [
+            {'event': 'slot', 'timestamp': None, 'name': 'public_search_response', 'value': "What is data science? Data science combines math and statistics, specialized programming, advanced analytics, artificial intelligence (AI), and machine learning with specific subject matter expertise to uncover actionable insights hidden in an organization's data. These insights can be used to guide decision making and strategic planning."},
+            {'event': 'slot', 'timestamp': None, 'name': 'kairon_action_response', 'value': "What is data science? Data science combines math and statistics, specialized programming, advanced analytics, artificial intelligence (AI), and machine learning with specific subject matter expertise to uncover actionable insights hidden in an organization's data. These insights can be used to guide decision making and strategic planning."}],
+            'responses': [
+                {'text': "What is data science? Data science combines math and statistics, specialized programming, advanced analytics, artificial intelligence (AI), and machine learning with specific subject matter expertise to uncover actionable insights hidden in an organization's data. These insights can be used to guide decision making and strategic planning.",
+                 'buttons': [], 'elements': [], 'custom': {}, 'template': None, 'response': None, 'image': None, 'attachment': None}]}
+
+
+def test_process_web_search_action_with_kairon_user_msg_entity():
+    action_name = "public_search_action_one_kairon_user_msg_entity"
+    bot = "5f51zd0a56b698ca10d35d2e"
+    user = 'test_user'
+    Actions(name=action_name, type=ActionType.web_search_action.value, bot=bot, user='test_user').save()
+    WebSearchAction(name=action_name, set_slot='public_search_response', top_n=1, bot=bot,
+                    user=user).save()
+
+    def _perform_web_search(*args, **kwargs):
```

Comment:
assert that the message sent in kairon_user_msg entity is actually sent for search